packages/react-environment/src/WebbProvider.tsx

```diff
@@ -51,31 +117,213 @@ export const WebbProvider: FC<WebbProviderProps> = ({ applicationName = 'Webb Da
+        interactiveFeedback.cancel();
+      }
+    });
+  };
+  const switchChain = async (chain: Chain, _wallet: Wallet) => {
+    const wallet = _wallet || activeWallet;
+    // wallet cleanup
+    if (_wallet.id !== activeWallet?.id && activeApi) {
+      await activeApi.destroy();
+    }
+    try {
+      setActiveChain(chain);
+      setActiveWallet(wallet);
+      let activeApi: WebbApiProvider<any> | null = null;
+      switch (wallet.id) {
+        case WalletsIds.Polkadot:
+          {
+            const url = chain.url;
+            const webbPolkadot = await WebbPolkadot.init('Webb DApp', [url], {
+              onError: (feedback) => {
+                registerInteractiveFeedback(setInteractiveFeedbacks, feedback);
+              },
+            });
+            setActiveApi(webbPolkadot);
+            activeApi = webbPolkadot;
+            setLoading(false);
+            const accounts = await webbPolkadot.accounts.accounts();
+            setAccounts(accounts);
+            await setActiveAccount(accounts[0]);
+            await webbPolkadot.accounts.setActiveAccount(accounts[0]);
+          }
+          break;
+        case WalletsIds.MetaMask:
+          {
+            const web3Provider = await Web3Provider.fromExtension();
+            const chainId = await web3Provider.network; // storage based on network id
+            const chainType = WebbWeb3Provider.storageName(chainId);
+            // const rinkebyS = await Storage.newFresh(WebbEVMChain.Rinkeby, rinkebyStorage);
+            // const mainS = await Storage.newFresh(WebbEVMChain.Main, mainStorage);
+            // const beresheetS = await Storage.newFresh(WebbEVMChain.Beresheet, beresheetStorage);
+            let storage: Storage<EVMStorage>;
+            switch (chainId) {
+              case WebbEVMChain.Main:
+                storage = await Storage.newFresh(chainType, mainStorage);
+                break;
+              case WebbEVMChain.Rinkeby:
+                storage = await Storage.newFresh(chainType, rinkebyStorage);
+                break;
+              case WebbEVMChain.Beresheet:
+                storage = await Storage.newFresh(chainType, beresheetStorage);
+                break;
+              default:
+                storage = await Storage.newFresh(chainType, beresheetStorage);
+                break;
+            }
+            const webbWeb3Provider = await WebbWeb3Provider.init(web3Provider, storage);
+            const accounts = await webbWeb3Provider.accounts.accounts();
+            setAccounts(accounts);
+            await setActiveAccount(accounts[0]);
+            await webbWeb3Provider.accounts.setActiveAccount(accounts[0]);
+            setActiveApi(webbWeb3Provider);
+            webbWeb3Provider.on('providerUpdate', async ([chainId]) => {
+              const nextChain = Object.values(chains).find((chain) => chain.evmId === chainId);
+              if (!nextChain) {
+                try {
```

Comment:
Should we be trying anything if we do not support the evm chain id?

Reply:
this call `WebbWeb3Provider.storageName(chainId)` will throw `UnsupportedChain` Error catching this and show an interactiveFeedback